examples/schema.rs

```diff
@@ -0,0 +1,137 @@
+use ion_schema::authority::{DocumentAuthority, FileSystemDocumentAuthority};
+use ion_schema::result::IonSchemaResult;
+use ion_schema::system::SchemaSystem;
+use std::fs;
+use std::path::Path;
+use std::str::from_utf8;
+
+fn main() -> IonSchemaResult<()> {
+    // load the YAML file which has all the CLI commands information stored in it
+    let yaml = load_yaml!("cli.yaml");
+    let matches = App::from_yaml(yaml).get_matches();
+    let (command_name, command_args) = matches.subcommand();
+    let command_args = command_args.unwrap();
+
+    match command_name {
+        "load" => load(command_args)?,
+        "validate" => validate(command_args)?,
+        _ => eprintln!("command name: {} not found", command_name),
+    }
+
+    Ok(())
+}
+
+fn load(command_args: &ArgMatches) -> IonSchemaResult<()> {
+    // Extract the user provided authorities
+    let authorities: Vec<_> = command_args.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = command_args.value_of("schema").unwrap();
+
+    // Set up authorities vector
+    let mut document_authorities: Vec<Box<dyn DocumentAuthority>> = vec![];
+
+    for authority in authorities {
+        document_authorities.push(Box::new(FileSystemDocumentAuthority::new(Path::new(
+            authority,
+        ))))
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    if schema.is_ok() {
+        eprintln!("Schema: {:?} was successfully loaded", schema.unwrap().id());
+    } else {
+        eprintln!("{:?}", schema.unwrap_err());
+    }
+
+    Ok(())
+}
+
+fn validate(command_args: &ArgMatches) -> IonSchemaResult<()> {
+    // Extract the user provided authorities
+    let authorities: Vec<_> = command_args.values_of("directories").unwrap().collect();
+
+    // Extract schema file provided by user
+    let schema_id = command_args.value_of("schema").unwrap();
+
+    // Extract Ion value provided by user
+    let input_file = command_args.value_of("input").unwrap();
+    let value = fs::read(input_file).expect("Can not load given ion file");
+    let owned_elements = element_reader()
+        .read_all(&value)
+        .expect("parsing failed unexpectedly");
+
+    // Set up authorities vector
+    let mut document_authorities: Vec<Box<dyn DocumentAuthority>> = vec![];
+
+    for authority in authorities {
+        document_authorities.push(Box::new(FileSystemDocumentAuthority::new(Path::new(
+            authority,
+        ))))
+    }
+
+    // Create a new schema system from given document authorities
+    let mut schema_system = SchemaSystem::new(document_authorities);
+
+    // load schema
+    let schema = schema_system.load_schema(schema_id);
+
+    // get the type defined within the schema file
+    // there will only be a single type defined within schema
```

Comment:
Please add this caveat to the `about` text for this subcommand. If/when we add this functionality to `ion beta schema` in the `ion-cli`, we should add a `--type` flag that allows the user to specify which type they want.

Reply:
added `--type` flag for validate sub command.